File: SRXLite/Services/BatchService.vb
Option Strict On

Imports System.Web
Imports System.Web.Services
Imports System.Web.Services.Protocols
Imports SRXLite.Classes
Imports SRXLite.DataTypes
Imports SRXLite.Modules

Namespace Web.Services

    <WebService(Namespace:="http://tempuri.org/srxlite/batchservice")> _
    <WebServiceBinding(ConformsTo:=WsiProfiles.BasicProfile1_1)> _
    <Global.Microsoft.VisualBasic.CompilerServices.DesignerGenerated()> _
    Public Class BatchService
        Inherits System.Web.Services.WebService

        Public _login As ServiceLogin
        Private _user As ServiceUser
        Private _batch As Batch
        Private _entity As Entity

#Region " Constructor "
        ''' <summary>
        ''' 
        ''' </summary>
        ''' <remarks></remarks>
        Public Sub New()
            _login = New ServiceLogin()
            _user = New ServiceUser()
        End Sub
#End Region

#Region " GetBatchUploadUrl "
        ''' <summary>
        ''' 
        ''' </summary>
        ''' <param name="batchType"></param>
        ''' <param name="entityName"></param>
        ''' <param name="docTypeID"></param>
        ''' <param name="location"></param>
        ''' <param name="callback"></param>
        ''' <param name="stateObject"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        <WebMethod()> _
        <SoapHeader("_login")> _
        Public Function BeginGetBatchUploadUrl( _
     ByVal batchType As BatchType, _
     ByVal location As String, _
     ByVal entityName As String, _
     ByVal docTypeID As Integer, _
     ByVal stylesheetUrl As String, _
     ByVal entityDisplayText As String, _
     ByVal callback As AsyncCallback, _
     ByVal stateObject As Object) As IAsyncResult

            Try
                _user.Authenticate(_login)
                _batch = New Batch(_user)
                Return _batch.BeginGetUploadUrl(batchType, location, entityName, docTypeID, stylesheetUrl, entityDisplayText, callback, stateObject)

            Catch accessEx As UnauthorizedAccessException
                LogError(accessEx.ToString)
                Throw accessEx
            Catch ex As Exception
                Dim soapEx As SoapException = CreateSoapException(ex.Message, ex.ToString, _user, "BeginGetBatchUploadUrl", True)
                Throw soapEx
            End Try
        End Function

        ''' <summary>
        ''' 
        ''' </summary>
        ''' <param name="result"></param>
        ''' <remarks></remarks>
        <WebMethod()> _
        <SoapHeader("_login")> _
        Public Function EndGetBatchUploadUrl(ByVal result As IAsyncResult) As String
            Try
                _user.Authenticate(_login)
                Return _batch.EndGetUploadUrl(result)

            Catch accessEx As UnauthorizedAccessException
                LogError(accessEx.ToString)
                Throw accessEx
            Catch ex As Exception
                Dim soapEx As SoapException = CreateSoapException(ex.Message, ex.ToString, _user, "EndGetBatchUploadUrl", True)
                Throw soapEx
            End Try
        End Function
#End Region

#Region " GetEntityBatchList "
        ''' <summary>
        ''' 
        ''' </summary>
        ''' <param name="entityName"></param>
        ''' <param name="callback"></param>
        ''' <param name="stateObject"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        <WebMethod()> _
        <SoapHeader("_login")> _
        Public Function BeginGetEntityBatchList( _
     ByVal entityName As String, _
     ByVal callback As AsyncCallback, _
     ByVal stateObject As Object) As IAsyncResult

            Try
                _user.Authenticate(_login)
                _entity = New Entity(_user, entityName)
                Return _entity.BeginGetBatchList(callback, stateObject)

            Catch accessEx As UnauthorizedAccessException
                LogError(accessEx.ToString)
                Throw accessEx
            Catch ex As Exception
                Dim soapEx As SoapException = CreateSoapException(ex.Message, ex.ToString, _user, "BeginGetEntityBatchList", True)
                Throw soapEx
            End Try
        End Function

        ''' <summary>
        ''' 
        ''' </summary>
        ''' <param name="result"></param>
        ''' <remarks></remarks>
        <WebMethod()> _
        <SoapHeader("_login")> _
        Public Function EndGetEntityBatchList(ByVal result As IAsyncResult) As List(Of BatchData)
            Try
                _user.Authenticate(_login)
                Return _entity.EndGetBatchList(result)

            Catch accessEx As UnauthorizedAccessException
                LogError(accessEx.ToString)
                Throw accessEx
            Catch ex As Exception
                Dim soapEx As SoapException = CreateSoapException(ex.Message, ex.ToString, _user, "EndGetEntityBatchList", True)
                Throw soapEx
            End Try
        End Function
#End Region


#Region " Dispose "
        Protected Overrides Sub Dispose(ByVal disposing As Boolean)
            If _batch IsNot Nothing Then _batch.Dispose()
            If _entity IsNot Nothing Then _entity.Dispose()
            MyBase.Dispose(disposing)
        End Sub
#End Region

    End Class

End Namespace

File: SRXLite.CS/Services/BatchService.cs
using System;
using System.Collections.Generic;
using System.Web.Services;
using System.Web.Services.Protocols;
using SRXLite.Classes;
using SRXLite.DataTypes;
using static SRXLite.Modules.ExceptionHandling;

namespace SRXLite.Web.Services
{
    [WebService(Namespace = "http://tempuri.org/srxlite/batchservice")]
    [WebServiceBinding(ConformsTo = WsiProfiles.BasicProfile1_1)]
    [System.ComponentModel.ToolboxItem(false)]
    public class BatchService : System.Web.Services.WebService
    {
        public ServiceLogin _login;
        private ServiceUser _user;
        private Batch _batch;
        private Entity _entity;

        #region Constructor

        /// <summary>
        /// Constructor
        /// </summary>
        public BatchService()
        {
            _login = new ServiceLogin();
            _user = new ServiceUser();
        }

        #endregion

        #region GetBatchUploadUrl

        [WebMethod]
        [SoapHeader("_login")]
        public IAsyncResult BeginGetBatchUploadUrl(
            BatchType batchType,
            string location,
            string entityName,
            int docTypeID,
            string stylesheetUrl,
            string entityDisplayText,
            AsyncCallback callback,
            object stateObject)
        {
            try
            {
                _user.Authenticate(_login);
                _batch = new Batch(_user);
                return _batch.BeginGetUploadUrl(batchType, location, entityName, docTypeID, stylesheetUrl, entityDisplayText, callback, stateObject);
            }
            catch (UnauthorizedAccessException accessEx)
            {
                LogError(accessEx.ToString());
                throw accessEx;
            }
            catch (Exception ex)
            {
                SoapException soapEx = CreateSoapException(ex.Message, ex.ToString(), _user, "BeginGetBatchUploadUrl", true);
                throw soapEx;
            }
        }

        [WebMethod]
        [SoapHeader("_login")]
        public string EndGetBatchUploadUrl(IAsyncResult result)
        {
            try
            {
                _user.Authenticate(_login);
                return _batch.EndGetUploadUrl(result);
            }
            catch (UnauthorizedAccessException accessEx)
            {
                LogError(accessEx.ToString());
                throw accessEx;
            }
            catch (Exception ex)
            {
                SoapException soapEx = CreateSoapException(ex.Message, ex.ToString(), _user, "EndGetBatchUploadUrl", true);
                throw soapEx;
            }
        }

        #endregion

        #region GetEntityBatchList

        [WebMethod]
        [SoapHeader("_login")]
        public IAsyncResult BeginGetEntityBatchList(
            string entityName,
            AsyncCallback callback,
            object stateObject)
        {
            try
            {
                _user.Authenticate(_login);
                _entity = new Entity(_user, entityName);
                return _entity.BeginGetBatchList(callback, stateObject);
            }
            catch (UnauthorizedAccessException accessEx)
            {
                LogError(accessEx.ToString());
                throw accessEx;
            }
            catch (Exception ex)
            {
                SoapException soapEx = CreateSoapException(ex.Message, ex.ToString(), _user, "BeginGetEntityBatchList", true);
                throw soapEx;
            }
        }

        [WebMethod]
        [SoapHeader("_login")]
        public List<BatchData> EndGetEntityBatchList(IAsyncResult result)
        {
            try
            {
                _user.Authenticate(_login);
                return _entity.EndGetBatchList(result);
            }
            catch (UnauthorizedAccessException accessEx)
            {
                LogError(accessEx.ToString());
                throw accessEx;
            }
            catch (Exception ex)
            {
                SoapException soapEx = CreateSoapException(ex.Message, ex.ToString(), _user, "EndGetEntityBatchList", true);
                throw soapEx;
            }
        }

        #endregion

        #region Dispose

        protected override void Dispose(bool disposing)
        {
            if (_batch != null) _batch.Dispose();
            if (_entity != null) _entity.Dispose();
            base.Dispose(disposing);
        }

        #endregion
    }
}
